lib/reek/cli/options.rb

```diff
@@ -213,6 +214,15 @@ def set_utility_options
           puts parser
           exit
         end
+        parser.on_tail('-l', '--list', 'List all available smell detectors') do
+          puts 'All available smell detectors:\n'
+          DetectorRepository.smell_types.map do |full_name|
+            full_name.to_s.split('::').last
+          end.each(&method(:puts))
```

Comment:
I'm not liking this way of printing an entire array. How about `puts the_array`?

Reply:
No problem, I can change that.